### ISSUE TITLE
http: fallback to GET if HEAD is not supported

### ISSUE THREAD

**Issue by efiop** [iterative/dvc#1909]
Reported by @robogeek.

 As @shcheklein noted, not all servers implement HEAD properly, so we could try to fallback to GET to check if file exists and get its etag. 

Steps to reproduce
```
(3.7.0-dvc) ➜  myrepo git:(master) ✗ dvc import https://data.cdc.gov/api/views/cjae-szjv/rows.csv air-quality.csv -v

DEBUG: '/home/efiop/git/dvc/myrepo/.dvc/updater' is outdated(
DEBUG: Trying to spawn '['/home/efiop/.pyenv/versions/3.7.0/envs/3.7.0-dvc/bin/python', '-m', 'dvc', 'daemon', '-q', '
updater']'
DEBUG: Spawned '['/home/efiop/.pyenv/versions/3.7.0/envs/3.7.0-dvc/bin/python', '-m', 'dvc', 'daemon', '-q', 'updater'
]'
DEBUG: PRAGMA user_version;
DEBUG: fetched: [(3,)]
DEBUG: CREATE TABLE IF NOT EXISTS state (inode INTEGER PRIMARY KEY, mtime TEXT NOT NULL, size TEXT NOT NULL, md5 TEXT
NOT NULL, timestamp TEXT NOT NULL)
DEBUG: CREATE TABLE IF NOT EXISTS state_info (count INTEGER)
DEBUG: CREATE TABLE IF NOT EXISTS link_state (path TEXT PRIMARY KEY, inode INTEGER NOT NULL, mtime TEXT NOT NULL)
DEBUG: INSERT OR IGNORE INTO state_info (count) SELECT 0 WHERE NOT EXISTS (SELECT * FROM state_info)
DEBUG: PRAGMA user_version = 3;
DEBUG: SELECT count from state_info WHERE rowid=1
DEBUG: fetched: [(8,)]
DEBUG: UPDATE state_info SET count = 8 WHERE rowid = 1
DEBUG: Path /home/efiop/git/dvc/myrepo/.dvc/cache inode 3676741
DEBUG: INSERT OR REPLACE INTO state(inode, size, mtime, timestamp, md5) VALUES (3676741, "20633", "1555675360886803456
", "1555783278390708992", "")
DEBUG: PRAGMA user_version;
DEBUG: fetched: [(3,)]
DEBUG: CREATE TABLE IF NOT EXISTS state (inode INTEGER PRIMARY KEY, mtime TEXT NOT NULL, size TEXT NOT NULL, md5 TEXT
NOT NULL, timestamp TEXT NOT NULL)
DEBUG: CREATE TABLE IF NOT EXISTS state_info (count INTEGER)
DEBUG: CREATE TABLE IF NOT EXISTS link_state (path TEXT PRIMARY KEY, inode INTEGER NOT NULL, mtime TEXT NOT NULL)
DEBUG: INSERT OR IGNORE INTO state_info (count) SELECT 0 WHERE NOT EXISTS (SELECT * FROM state_info)
DEBUG: PRAGMA user_version = 3;
DEBUG: Removing output 'air-quality.csv' of 'air-quality.csv.dvc'.
DEBUG: Removing 'air-quality.csv'
Importing 'https://data.cdc.gov/api/views/cjae-szjv/rows.csv' -> '/home/efiop/git/dvc/myrepo/air-quality.csv'
DEBUG: Computed stage 'air-quality.csv.dvc' md5: '0b1341766eb8f48dcf3b8f78add5a563'
DEBUG: Downloading 'https://data.cdc.gov/api/views/cjae-szjv/rows.csv' to '/home/efiop/git/dvc/myrepo/air-quality.csv'
[##############################] 100% air-quality.csv
DEBUG: SELECT count from state_info WHERE rowid=1
DEBUG: fetched: [(8,)]
DEBUG: UPDATE state_info SET count = 8 WHERE rowid = 1
DEBUG: Path /home/efiop/git/dvc/myrepo/.dvc/cache inode 3676741
DEBUG: INSERT OR REPLACE INTO state(inode, size, mtime, timestamp, md5) VALUES (3676741, "20633", "1555675360886803456
", "1555783286015181056", "")
ERROR: failed to import https://data.cdc.gov/api/views/cjae-szjv/rows.csv. You could also try downloading it manually
and adding it with `dvc add` command. - dependency 'https://data.cdc.gov/api/views/cjae-szjv/rows.csv' does not exist
------------------------------------------------------------
Traceback (most recent call last):
  File "/home/efiop/.pyenv/versions/3.7.0/envs/3.7.0-dvc/lib/python3.7/site-packages/dvc/command/imp.py", line 22, in
run
    self.repo.imp(self.args.url, out, self.args.resume)
  File "/home/efiop/.pyenv/versions/3.7.0/envs/3.7.0-dvc/lib/python3.7/site-packages/dvc/repo/scm_context.py", line 4,
 in run
    result = method(repo, *args, **kw)
  File "/home/efiop/.pyenv/versions/3.7.0/envs/3.7.0-dvc/lib/python3.7/site-packages/dvc/repo/imp.py", line 17, in imp
    stage.run(resume=resume)
  File "/home/efiop/.pyenv/versions/3.7.0/envs/3.7.0-dvc/lib/python3.7/site-packages/dvc/stage.py", line 805, in run
    self.save()
  File "/home/efiop/.pyenv/versions/3.7.0/envs/3.7.0-dvc/lib/python3.7/site-packages/dvc/stage.py", line 660, in save
    dep.save()
  File "/home/efiop/.pyenv/versions/3.7.0/envs/3.7.0-dvc/lib/python3.7/site-packages/dvc/output/base.py", line 173, in
 save
    raise self.DoesNotExistError(self)
dvc.dependency.base.DependencyDoesNotExistError: dependency 'https://data.cdc.gov/api/views/cjae-szjv/rows.csv' does not exist
```

Related discussion on discord: https://discordapp.com/channels/485586884165107732/485586884165107734/568946737692147734

**Comment by alustenberg** [comment]
seems this was resolved as part of #4131, ef944e650

```
(.env) dvc-test/ $ curl -I https://data.cdc.gov/api/views/cjae-szjv/rows.csv                          
HTTP/1.1 400 Bad Request
Server: nginx
Date: Wed, 02 Dec 2020 15:37:40 GMT
Connection: keep-alive
Access-Control-Allow-Origin: *
X-Error-Code: invalid_request
X-Error-Message: HEAD is not supported
Cache-Control: private, no-cache, must-revalidate
Age: 0
X-Socrata-Region: aws-us-east-1-fedramp-prod
Strict-Transport-Security: max-age=31536000; includeSubDomains
X-Socrata-RequestId: 15b271931ecb9614478b3e06b5080ec7

(.env) dvc-test/ $ dvc import-url https://data.cdc.gov/api/views/cjae-szjv/rows.csv air-quality.csv -v
2020-12-02 10:34:32,111 DEBUG: Check for update is enabled.
2020-12-02 10:34:32,112 DEBUG: fetched: [(3,)]                        
2020-12-02 10:34:32,258 DEBUG: Removing output 'air-quality.csv' of stage: 'air-quality.csv.dvc'.
Importing 'https://data.cdc.gov/api/views/cjae-szjv/rows.csv' -> 'air-quality.csv'
2020-12-02 10:34:32,258 DEBUG: Computed stage: 'air-quality.csv.dvc' md5: 'a6d21f5d9c2292d92e8982f4765d1931'
2020-12-02 10:34:32,258 DEBUG: 'md5' of stage: 'air-quality.csv.dvc' changed.
2020-12-02 10:34:33,839 DEBUG: Downloading 'https://data.cdc.gov/api/views/cjae-szjv/rows.csv' to 'air-quality.csv'
2020-12-02 10:34:41,638 DEBUG: Adding 'air-quality.csv' to '.gitignore'.                                                                                                                       
2020-12-02 10:34:41,639 DEBUG: Path '/home/alexl/src/dvc-test/air-quality.csv' inode '29626617'
2020-12-02 10:34:41,639 DEBUG: fetched: []
2020-12-02 10:34:41,715 DEBUG: Path 'air-quality.csv' inode '29626617'
2020-12-02 10:34:41,715 DEBUG: fetched: []
2020-12-02 10:34:41,716 DEBUG: {'air-quality.csv': 'modified'}
2020-12-02 10:34:41,716 DEBUG: Path '/home/alexl/src/dvc-test/air-quality.csv' inode '29626617'
2020-12-02 10:34:41,716 DEBUG: fetched: [('1606923280326207488', '49575093', 'cf8ba105a3dd78bc6a5f9f09c4c616e2', '1606923281716043008')]
2020-12-02 10:34:41,716 DEBUG: Computed stage: 'air-quality.csv.dvc' md5: '4824ff9ed03b8c89d33d5b519008598c'
2020-12-02 10:34:41,717 DEBUG: Saving 'air-quality.csv' to '.dvc/cache/cf/8ba105a3dd78bc6a5f9f09c4c616e2'.
2020-12-02 10:34:41,717 DEBUG: cache '/home/alexl/src/dvc-test/.dvc/cache/cf/8ba105a3dd78bc6a5f9f09c4c616e2' expected 'HashInfo(name='md5', value='cf8ba105a3dd78bc6a5f9f09c4c616e2', dir_info=None, size=49575093, nfiles=None)' actual 'None'
2020-12-02 10:34:41,717 DEBUG: Cache type 'reflink' is not supported: reflink is not supported
2020-12-02 10:34:41,737 DEBUG: Created 'copy': .dvc/cache/cf/8ba105a3dd78bc6a5f9f09c4c616e2 -> air-quality.csv
2020-12-02 10:34:41,737 DEBUG: Path 'air-quality.csv' inode '29626619'
2020-12-02 10:34:41,737 DEBUG: Path 'air-quality.csv' inode '29626619'
2020-12-02 10:34:41,737 DEBUG: fetched: []
2020-12-02 10:34:41,737 DEBUG: Path '.dvc/cache/cf/8ba105a3dd78bc6a5f9f09c4c616e2' inode '29626617'
2020-12-02 10:34:41,737 DEBUG: fetched: [('1606923280326207488', '49575093', 'cf8ba105a3dd78bc6a5f9f09c4c616e2', '1606923281716412672')]
2020-12-02 10:34:41,738 DEBUG: Saving information to 'air-quality.csv.dvc'.

To track the changes with git, run:

	git add air-quality.csv.dvc .gitignore
2020-12-02 10:34:41,740 DEBUG: fetched: [(0,)]
2020-12-02 10:34:41,747 DEBUG: Analytics is enabled.
2020-12-02 10:34:41,779 DEBUG: Trying to spawn '['daemon', '-q', 'analytics', '/tmp/tmpep3nxa6a']'
2020-12-02 10:34:41,779 DEBUG: Spawned '['daemon', '-q', 'analytics', '/tmp/tmpep3nxa6a']'
```

**Comment by skshetry** [comment]
Thanks for reminding @alustenberg. That was fixed already.
